src/test/helpers/approval/approval-helper-s.test.js

```diff
@@ -0,0 +1,153 @@
+    });
+  });
+
+  afterEach(() => {
+    getSpy.mockReset();
+    unlinkWorkflowsSpy.mockReset();
+    linkWorkflowSpy.mockReset();
+  });
+  it('should build correct url for workflows initial lookup query', () => {
+    mockApi
+      .onGet(
+        `${APPROVAL_API_BASE}/workflows?app_name=catalog&object_type=Portfolio&object_id=123&filter[name][contains]=&page_size=50&page=1`
+      )
+      .reply(200, {
+        results: [
+          {
+            name: 'workflow1',
+            id: '111'
+          }
+        ]
+      });
+
+    return ApprovalHelper.loadWorkflowOptions('some-filter-value', [
+      'initial-id-1',
+      'initial-id-2'
+    ]).then(() => {
+      const expectedFilterFragment =
+        '&filter[id][]=initial-id-1&filter[id][]=initial-id-2';
+      expect(getSpy).toHaveBeenCalledWith(
+        `${APPROVAL_API_BASE}/workflows?name=some-filter-value&id=initial-id-1&id=initial-id-2`
+      );
+    });
+  });
+
+  it('should call link and unlink workflows with correct arguments', () => {
+    mockApi.onGet(`${CATALOG_API_BASE}/me/`).replyOnce(200, {
+      username: 'fred',
+      first_name: 'Fred',
+      last_name: 'Flintstone'
+    });
+    mockApi
+      .onPost(`${APPROVAL_API_BASE}/workflows/unlink-id-2/unlink/`)
+      .replyOnce((req) => {
+        expect(JSON.parse(req.data)).toEqual({
+          object_type: 'Portfolio',
+          app_name: 'catalog',
+          object_id: '5'
+        });
+        return [204];
+      });
+
+    mockApi
+      .onPost(`${APPROVAL_API_BASE}/workflows/unlink-id-1/unlink/`)
+      .replyOnce((req) => {
+        expect(JSON.parse(req.data)).toEqual({
+          object_type: 'Portfolio',
+          app_name: 'catalog',
+          object_id: '5'
+        });
+        return [204];
+      });
+    mockApi
+      .onPost(`${APPROVAL_API_BASE}/workflows/link-id-1/link/`)
+      .replyOnce((req) => {
+        expect(JSON.parse(req.data)).toEqual({
+          object_type: 'Portfolio',
+          app_name: 'catalog',
+          object_id: '5'
+        });
+        return [204];
+      });
+    mockApi
+      .onPost(`${APPROVAL_API_BASE}/workflows/link-id-2/link/`)
+      .replyOnce((req) => {
+        expect(JSON.parse(req.data)).toEqual({
+          object_type: 'Portfolio',
+          app_name: 'catalog',
+          object_id: '5'
+        });
+        return [204];
+      });
+
+    return ApprovalHelper.updateWorkflows(
+      ['unlink-id-1', 'unlink-id-2'],
+      ['link-id-1', 'link-id-2'],
+      {
+        object_type: 'Portfolio',
+        app_name: 'catalog',
+        object_id: '5'
+      }
+    ).then((data) => {
+      //expect(unlinkWorkflowsSpy).toHaveBeenNthCalledWith(1, [
+      //  'unlink-id-1',
+      //  {}
+      //]);
+      //expect(unlinkWorkflowsSpy).toHaveBeenNthCalledWith(2, 'unlink-id-2', {});
+      //expect(linkWorkflowSpy).toHaveBeenNthCalledWith(3, 'link-id-1', {});
+      //expect(linkWorkflowSpy).toHaveBeenNthCalledWith(4, 'link-id-2', {});
```

Comment:
Are these just test expectations or does the `toHaveBeenNthCalledWith` not work as we want it to or something like that? I think if you want to leave these in as comments for bolstering this test later, that's probably fine, but we should have at least one expectation that gets asserted here, otherwise this test "passes" but only because there's nothing to fail on.

Reply:
@eclarizio - forgot to uncomment these afdter fixing the mock posts - test updated now